corehq/apps/repeaters/models.py

```diff
@@ -175,6 +168,24 @@ def get_payload(self, repeat_record):
         generator = self.get_payload_generator(self.format_or_default_format())
         return generator.get_payload(repeat_record, self.payload_doc(repeat_record))
 
+    def get_payload_and_handle_exception(self, repeat_record, save_failure=True):
+        try:
+            return self.get_payload(repeat_record)
+        except ResourceNotFound as e:
```

Comment:
Yeah a lot of these checks seemed pretty arbitrary and possibly unnecessary at this point. I limited myself to trying to just keep things equivalent though, and also make it clearer what exceptions are caught where so it's easier to reason with (and maybe do something about later)